js-miniapp-bridge/src/common-bridge.ts

```diff
@@ -0,0 +1,120 @@
+/* tslint:disable:no-any */
+const mabMessageQueue: Callback[] = [];
+export { mabMessageQueue };
+
+export interface Callback {
+  id: string;
+
+  onSuccess: (value: string) => void;
+  onError: (error: string) => void;
+}
+
+export class MiniAppBridge {}
+
+export interface MiniAppBridge {
```

Comment:
I think we don't need to define the `MiniAppBridge` interface and class separately. You can define the typing at the same time as defining the class, and also the method return values can be inferred by Typescript.

The only problem is that the `exec` function needs to be implemented by the platform specific code. So maybe you could create a separate `PlatformExecutor` which is provided when constructing the `MiniAppBridge`?

Something like this in `common-bridge.ts`:
```
export interface PlatformExecutor {
  exec(
    action: string,
    param: any,
    onSuccess: (value: string) => void,
    onError: (error: string) => void
  ): void;
}

export class MiniAppBridge {
  executor: PlatformExecutor;

  constructor(executor: PlatformExecutor) {
    this.executor = executor;
  }

  requestPermission(permissionType: string) {
    return new Promise<string>((resolve, reject) => {
      return this.executor.exec(
        // ...etc
      );
    });
  }

  // ...etc
}
```

And then implement it in `android-bridge.ts`:
```
class AndroidExcecutor implements PlatformExecutor {
  exec(action, param, onSuccess, onError) {
      // ...
    );
  }
}

(window as any).MiniAppBridge = new MiniAppBridge(new AndroidExcecutor());
```

What do you think?

Reply:
I think good to go that way.